core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessage.java

```diff
@@ -145,16 +149,23 @@ static void failLateSubscriber(SubscriptionImpl subscription, Subscriber<?> late
         final Throwable cause = abortedOrLate(oldSubscriber);
 
         if (subscription.needsDirectInvocation()) {
-            lateSubscriber.onSubscribe(NoopSubscription.INSTANCE);
-            lateSubscriber.onError(cause);
+            lateSubscriber(lateSubscriber, cause);
         } else {
             subscription.executor().execute(() -> {
-                lateSubscriber.onSubscribe(NoopSubscription.INSTANCE);
-                lateSubscriber.onError(cause);
+                lateSubscriber(lateSubscriber, cause);
             });
         }
     }
 
+    private static void lateSubscriber(Subscriber<?> lateSubscriber, Throwable cause) {
+        try {
+            lateSubscriber.onSubscribe(NoopSubscription.INSTANCE);
+            lateSubscriber.onError(cause);
+        } catch (Exception e) {
```

Comment:
Global comment: Should we catch a `Throwable`?

Reply:
https://github.com/reactive-streams/reactive-streams-jvm/issues/107#issuecomment-54142644
I think they are guiding not to catch the fatal ones. 🤔

Reply:
I think there are `Error` that aren't fatal, so better to catch `Throwable` and we may need a `propagateIfFatal` like in zipkin

```java
// Taken from RxJava throwIfFatal, which was taken from scala
  public static void propagateIfFatal(Throwable t) {
    if (t instanceof VirtualMachineError) {
      throw (VirtualMachineError) t;
    } else if (t instanceof ThreadDeath) {
      throw (ThreadDeath) t;
    } else if (t instanceof LinkageError) {
      throw (LinkageError) t;
    }
  }
```

Reply:
Ah, let me add this method to `Exceptions`. Thanks!

Reply:
I think we don't have to follow the Reactive Streams TCK, just FYI, which regards `InterruptedException` as fatal. 
https://github.com/reactive-streams/reactive-streams-jvm/blob/master/tck/src/main/java/org/reactivestreams/tck/flow/support/NonFatal.java#L35
It is copied from Scala code.
https://github.com/scala/scala/blob/v2.13.1/src/library/scala/util/control/NonFatal.scala#L41

Reply:
Ah yes, I saw it from https://github.com/ReactiveX/RxJava/issues/748#issuecomment-32471495
and I think they intentionally omit that. But I didn't think about it deeply. 😆